### PR TITLE
This is a template for building a scikit-learn-based machine learning trainer  that can be run on AI Platform. 

### PR DESCRIPTION
**Pull request by luotigerlsx**
This is a template for building a scikit-learn-based machine learning trainer 
that can be run on AI Platform. By using the template, the user can easily train and deploy their scikit-learn-based model using AI Platform.

The structure of the template is as follows:
```
template 
    |__ config
        |__ config.yaml             # for running normal training job on AI Platform
        |__ hptuning_config.yaml    # for running hyperparameter tunning job on AI Platform    
    |__ scripts
        |__ train.sh                # convenience script for running machine learning training jobs
        |__ deploy.sh               # convenience script for deploying trained scikit-learn model
        |__ predict.sh              # convenience script for requesting online prediction
        |__ predict.py              # helper function for requesting online prediction using python
    |__ trainer                     # trainer package
        |__ metadata.py             # dataset metadata and feature columns definitions
        |__ model.py                # pre-processing and machine learning model pipeline definition
        |__ utils.py                # utility functions including e.g. loading data from bigquery and cloud storage
        |__ task.py                 # training job entry point, handling the parameters passed from command line 
    |__ setup.py                    # specify necessary dependency for running job on AI Platform
    |__ requirements.txt            # specify necessary dependency, helper for setup environemnt for local development
```

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/cloudml-samples/400)

### CODE REVIEW

File: sklearn/sklearn-template/template/README.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+CSV_COLUMNS = None  # Schema of the data. Necessary for data stored in GCS
+
+NUMERIC_FEATURES = [
+    'sepal_length',
+    'sepal_width',
+    'petal_length',
+    'petal_width',
+]
+
+# Fill this with any categorical features in the dataset
```
Comment (gogasca) [inline sklearn/sklearn-template/template/README.md]: I think would be relevant to add a sample using both categorical and numerical in the future.

File: sklearn/sklearn-template/template/README.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+- `hptuning_config.yaml`: for running hyperparameter tuning job on AI Platform.
+
+The YAML files share some configuration parameters. In particular, `runtimeVersion` and `pythonVersion` should
+correspond in both files.
+
+```yaml
+trainingInput:
+  scaleTier: STANDARD_1   # Machine type
+  runtimeVersion: "1.13"  # Scikit-learn version
+  pythonVersion: "2.7"    # only support python 2.7 and 3.5
```
Comment (gogasca) [inline sklearn/sklearn-template/template/README.md]: nit: Use Python 3.5 as default.

File: sklearn/sklearn-template/template/config/config.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+trainingInput:
+  scaleTier: STANDARD_1
+  runtimeVersion: "1.13"
+  pythonVersion: "2.7"  # Note: Python 3 is also supported
```
Comment (gogasca) [inline sklearn/sklearn-template/template/config/config.yaml]: nit: Same as above

File: sklearn/sklearn-template/template/README.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+- VERSION_NAME: Version of the model to be deployed`.
+
+**Note**: Please make sure the following parameters are properly set in deploy.sh 
+```shell
+REGION=us-central1
+
+# The following two parameters should be aligned with those used during
+# training job, i.e., specified in the yaml files under config/
+RUN_TIME=1.13
+PYTHON_VERSION=2.7 # only support python 2.7 and 3.5
```
Comment (gogasca) [inline sklearn/sklearn-template/template/README.md]: nit: Same as above

File: sklearn/sklearn-template/template/config/hptuning_config.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+trainingInput:
+  scaleTier: STANDARD_1
+  runtimeVersion: "1.13"
+  pythonVersion: "2.7"  # Note: Python 3 is also supported
```
Comment (gogasca) [inline sklearn/sklearn-template/template/config/hptuning_config.yaml]: nit: Same as above

File: sklearn/sklearn-template/template/scripts/deploy.sh [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+#   $3: Version of the model to be deployed
+
+MODEL_DIR=$1
+MODEL_NAME=$2
+VERSION_NAME=$3
+
+REGION=us-central1
+FRAMEWORK=SCIKIT_LEARN
+RUN_TIME=1.13
+PYTHON_VERSION=2.7 # only support python 2.7 and 3.5
```
Comment (gogasca) [inline sklearn/sklearn-template/template/scripts/deploy.sh]: nit: Same as above

**Comment by cfezequiel** [comment]
Thanks for the feedback Gonzalo. Will address the comments.

File: sklearn/sklearn-template/template/README.md (hunk 1 above)
Comment (cfezequiel) [reply sklearn/sklearn-template/template/README.md]: Agreed. I've added a TODO comment in trainer/metadata.py. I'm currently working on getting an ML dataset with categorical variables uploaded in BQ public datasets.

File: sklearn/sklearn-template/template/README.md (hunk 2 above)
Comment (cfezequiel) [reply sklearn/sklearn-template/template/README.md]: Done.

File: sklearn/sklearn-template/template/config/config.yaml (hunk 3 above)
Comment (cfezequiel) [reply sklearn/sklearn-template/template/config/config.yaml]: Done.

File: sklearn/sklearn-template/template/config/hptuning_config.yaml (hunk 5 above)
Comment (cfezequiel) [reply sklearn/sklearn-template/template/config/hptuning_config.yaml]: Done.

File: sklearn/sklearn-template/template/scripts/deploy.sh (hunk 6 above)
Comment (cfezequiel) [reply sklearn/sklearn-template/template/scripts/deploy.sh]: Done.

**Comment by cfezequiel** [comment]
I've pushed changes to the code based on the comments.

**Comment by gogasca** [comment]
LGTM

**Pull request merged**